src/game/common/system/asciistring.cpp

```diff
@@ -203,13 +212,13 @@ char *Utf8String::Get_Buffer_For_Read(int len)
 /**
  * Set this string content to a new string
  */
-void Utf8String::Set(const char *c)
+void Utf8String::Set(const char *str)
 {
-    if (m_data == nullptr || c != m_data->Peek()) {
-        size_t len;
+    if (m_data == nullptr || str != m_data->Peek()) {
+        const size_t len = str ? strlen(str) : 0;
 
-        if (c && (len = strlen(c) + 1, len != 1)) {
```

Comment:
Looks like it was written this way to reduce branch count by 1. But it is difficult to read. I changed it so it compares better with original code.